### ISSUE TITLE
Graph Duplication Bug (delete graph also stops working)

### ISSUE THREAD

**Issue by shishir03** [flapjs/vanilla#48]
Attempting to change the type of graph (for instance, from a PDA to an NFA) while the add edge dialog box is still open results in the dialog box still showing up in the "new" graph, as shown below.

file:///home/shishiriyer/Pictures/Screenshots/Screenshot%20from%202023-05-12%2020-13-50.png
file:///home/shishiriyer/Pictures/Screenshots/Screenshot%20from%202023-05-12%2020-13-59.png

While the box goes away if you move any of the edges on the new graph, deleting the graph does not clear the screen if there was a previous graph, and instead reverts to this previous graph. Pressing the delete button on any other graph type also duplicates this graph as shown below.

file:///home/shishiriyer/Pictures/Screenshots/Screenshot%20from%202023-05-12%2020-17-32.png

Meanwhile, pressing "delete" on the dialog box results in the graph from the previous machine type getting duplicated, overwriting the old graph. Pressing the delete key on this old graph does not work either.

file:///home/shishiriyer/Pictures/Screenshots/Screenshot%20from%202023-05-12%2020-15-47.png

This probably has to do with some weird stuff with the history stack.

**Comment by Lexseal** [comment]
What did you mean by [deleting the graph](https://github.com/flapjs/vanilla/issues/48#:~:text=the%20new%20graph%2C-,deleting%20the%20graph,-does%20not%20clear)? I just tried it out and it seemed after you move the new graph around to make the dialog go away, all the functionalities preceded as normal.

**Comment by Lexseal** [comment]
When you say delete key, did you mean dd? I think that might be a separate bug where keybinding was removed for some reason.

**Comment by shishir03** [comment]
> What did you mean by [deleting the graph](https://github.com/flapjs/vanilla/issues/48#:~:text=the%20new%20graph%2C-,deleting%20the%20graph,-does%20not%20clear)? I just tried it out and it seemed after you move the new graph around to make the dialog go away, all the functionalities preceded as normal.

Yeah moving the graph around makes things go back to normal. Stuff breaks if you hit the dd key though (which otherwise works normally).

**Comment by Lexseal** [comment]
Issue most likely caused by setting `graph=consts.EMPTY_GRAPH` instead of `{}`. Note when we modify the graph, we also modify `consts.EMPTY_GRAPH`.

It is tentatively fixed now.
